code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -57,6 +64,80 @@
 	adjacent_turfs.Cut()
 	return ..()
 
+///adds a gas or list of gases to our filter_types. used so that the scrubber can check if its supposed to be processing after each change
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/add_filters(filter_or_filters)
+	if(!islist(filter_or_filters))
+		filter_or_filters = list(filter_or_filters)
+
+	for(var/gas_to_filter in filter_or_filters)
+		var/translated_gas = istext(gas_to_filter) ? gas_id2path(gas_to_filter) : gas_to_filter
+
+		if(ispath(translated_gas, /datum/gas))
+			filter_types |= translated_gas
+			continue
+
+	var/turf/open/our_turf = get_turf(src)
+
+	if(!isopenturf(our_turf))
+		return FALSE
+
+	var/datum/gas_mixture/turf_gas = our_turf.air
+	if(!turf_gas)
+		return FALSE
+
+	check_atmos_process(our_turf, turf_gas, turf_gas.temperature)
+	return TRUE
+
+///remove a gas or list of gases from our filter_types.used so that the scrubber can check if its supposed to be processing after each change
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/remove_filters(filter_or_filters)
+	if(!islist(filter_or_filters))
+		filter_or_filters = list(filter_or_filters)
+
+	for(var/gas_to_filter in filter_or_filters)
+		var/translated_gas = istext(gas_to_filter) ? gas_id2path(gas_to_filter) : gas_to_filter
+
+		if(ispath(translated_gas, /datum/gas))
+			filter_types -= translated_gas
+			continue
+
+	var/turf/open/our_turf = get_turf(src)
+	var/datum/gas_mixture/turf_gas
+
+	if(isopenturf(our_turf))
+		turf_gas = our_turf.air
+
+	if(!turf_gas)
+		return FALSE
```

Comment:
```suggestion
	var/datum/gas_mixture/turf_gas

	if(isopenturf(our_turf))
		turf_gas = our_turf.air

	if(!isopenturf(our_turf))
		return FALSE
	var/datum/gas_mixture/turf_gas = our_turf.air
```

Reply:
???

Reply:
okay, looking more deeply into this, our_turf is a /turf/open variable, but the assigned turf may not always be /turf/open and the air variable doesn't exist on turfs that aren't /turf/open, so this piece of code seemed to be doing practices that seemed redundant, which is not the case looking at it now.